kie-server-parent/kie-server-services/kie-server-services-jbpm-queries/src/main/java/org/kie/server/services/jbpm/queries/JbpmQueriesKieServerExtension.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.services.jbpm.queries;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+import org.jbpm.services.api.query.QueryService;
+import org.kie.server.api.KieServerConstants;
+import org.kie.server.services.api.KieContainerInstance;
+import org.kie.server.services.api.KieServerApplicationComponentsService;
+import org.kie.server.services.api.KieServerExtension;
+import org.kie.server.services.api.KieServerRegistry;
+import org.kie.server.services.api.SupportedTransports;
+import org.kie.server.services.impl.KieServerImpl;
+import org.kie.server.services.jbpm.JbpmKieServerExtension;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JbpmQueriesKieServerExtension implements KieServerExtension {
+
+    public static final String EXTENSION_NAME = "jBPMQueries";
+    
+    private static final Logger logger = LoggerFactory.getLogger(JbpmQueriesKieServerExtension.class);
+
+    private static final Boolean disabled = Boolean.parseBoolean(System.getProperty(KieServerConstants.KIE_JBM_QUERIES_SERVER_EXT_DISABLED, "false"));
+
+    private KieServerImpl kieServer;
+    private KieServerRegistry context;
+    
+    private QueryService queryService;
+    
+    private List<Object> services = new ArrayList<Object>();
+    private boolean initialized = false;
+
+    @Override
+    public boolean isInitialized() {
+        return initialized;
+    }
+
+    @Override
+    public boolean isActive() {
+        return disabled == false;
```

Comment:
since this requires jbpm extension would be worth checking if it was not disabled - similar how jbpm-ui does it